typing/typecore.ml

```diff
@@ -5749,7 +5808,7 @@ and type_expect_
         exp_type = type_constant cst;
         exp_attributes = sexp.pexp_attributes;
         exp_env = env }
-  | Pexp_let(Nonrecursive,
+  | Pexp_let(_, Nonrecursive,
```

Comment:
I think we should match explicitly on `Immutable` here.  It's already true that this case will never fire for let mutable, because of invariants on what patterns we allow with `let mutable` and the definition of `turn_let_into_match`.  But it would be a bug if this transformation `did` happen to let mutable, so better to be explicit about it.